[Task] Add input validation for the application id to CLI AsyncAPI

**Is your task related to a problem? Please describe.**

Currently, users can specify a string without a colon as a value for the `--application-id` option of the `aspect to asyncapi` subcommand. It generates an AsyncAPI specification without any errors.

```
$ java -jar tools/samm-cli/target/samm-cli-DEV-SNAPSHOT.jar aspect https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl to asyncapi -o /tmp/bpn.yaml --application-id=bpn-app
$ cat /tmp/bpn.yaml
---
asyncapi: 3.0.0
info:
  title: Shared Aspect for any business partner number MQTT API
  version: v2
  description: This is a shared aspect for BPN with a regex.
  x-samm-aspect-model-urn: "urn:samm:io.catenax.shared.business_partner_number:2.0.0#BusinessPartnerNumber"
defaultContentType: application/json
channels:
  BusinessPartnerNumber:
    address: /io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber
    description: This channel for updating BusinessPartnerNumber Aspect.
    parameters:
      namespace: io.catenax.shared.business_partner_number
      version: 2.0.0
      aspect-name: BusinessPartnerNumber
    messages: {}
operations: {}
components:
  messages: {}
  schemas: {}
id: bpn-app
```

But running the official AsyncAPI tools against this spec causes an error, as follows:

```
$ npm i -g @asyncapi/cli
$ asyncapi validate /tmp/bpn.yaml 

File /tmp/bpn.yaml and/or referenced documents have governance issues.
Errors 
/tmp/bpn.yaml
 22:5  error  asyncapi-document-resolved  "id" property must match format "uri"  id

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)

$ asyncapi generate fromTemplate /tmp/bpn.yaml @asyncapi/html-template --output ./docs
┌  AsyncAPI Generator
│
◐  Generation in progress. Keep calm and wait a bit..Diagnostic err: The latest version of AsyncAPi is not used. It is recommended update to the "3.1.0" version. in path ["asyncapi"] starting L2 C10, ending L2 C15
Diagnostic err: "id" property must match format "uri" in path ["id"] starting L22 C4, ending L22 C11
Error: Input is not a correct AsyncAPI document so it cannot be processed.
    at Generator.parseInput (/home/sekikn/.nvm/versions/node/v22.20.0/lib/node_modules/@asyncapi/cli/node_modules/@asyncapi/generator/lib/generator.js:373:21)
    at async Generator.configureTemplateWorkflow (/home/sekikn/.nvm/versions/node/v22.20.0/lib/node_modules/@asyncapi/cli/node_modules/@asyncapi/generator/lib/generator.js:308:5)
    at async Generator.generate (/home/sekikn/.nvm/versions/node/v22.20.0/lib/node_modules/@asyncapi/cli/node_modules/@asyncapi/generator/lib/generator.js:197:5)
    at async Generator.generateFromString (/home/sekikn/.nvm/versions/node/v22.20.0/lib/node_modules/@asyncapi/cli/node_modules/@asyncapi/generator/lib/generator.js:444:12) {
  diagnostics: [
    {
      code: 'asyncapi-latest-version',
      message: 'The latest version of AsyncAPi is not used. It is recommended update to the "3.1.0" version.',
      path: [Array],
      severity: 2,
      range: [Object],
      documentationUrl: undefined
    },
    {
      code: 'asyncapi-document-resolved',
      message: '"id" property must match format "uri"',
      path: [Array],
      severity: 0,
      range: [Object],
      documentationUrl: undefined
    }
  ]
}
◇  Generation failed
Generator Error: Input is not a correct AsyncAPI document so it cannot be processed.
```

This is because the "id" property must comply with the URI format defined in RFC3986, [as the official documentation says](https://www.asyncapi.com/docs/reference/specification/v3.0.0#A2SIdString). It seems to imply that the URI here must be an absolute URI (the scheme + ":" + hier-part format), and not a relative URI (without a colon).

**Describe the solution you'd like**

I suppose it would be better if SAMM-CLI rejects a relative URI in the input validation, since it helps users avoid generating an invalid spec and discovering problems later.

**Describe alternatives you've considered**

No specific alternatives for this issue, as the proposed solution is simple and straightforward.

**Additional context**

No additional context.

--- a/tools/samm-cli/src/main/java/org/eclipse/esmf/aspect/to/AspectToAsyncapiCommand.java
+++ b/tools/samm-cli/src/main/java/org/eclipse/esmf/aspect/to/AspectToAsyncapiCommand.java
@@ -17,6 +17,7 @@ import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.util.Locale;
@@ -76,7 +77,7 @@ public class AspectToAsyncapiCommand extends AbstractCommand {
    @CommandLine.Option(
          names = { "--application-id", "-ai" },
          description = "Use this param for provide application id." )
-   private String applicationId;
+   private URI applicationId;
 
    @CommandLine.Option(
          names = { "--channel-address", "-ca" },
@@ -112,6 +113,10 @@ public class AspectToAsyncapiCommand extends AbstractCommand {
 
    @Override
    public void run() {
+      if ( applicationId != null && !applicationId.isAbsolute() ) {
+         throw new CommandException( "Application id must be an absolute URI." );
+      }
+
       setDetails( details );
       setResolverConfig( resolverConfiguration );
 
@@ -119,7 +124,7 @@ public class AspectToAsyncapiCommand extends AbstractCommand {
       final Aspect aspect = getInputHandler( parentCommand.parentCommand.getInput() ).loadAspect();
       final AsyncApiSchemaGenerationConfig config = AsyncApiSchemaGenerationConfigBuilder.builder()
             .useSemanticVersion( useSemanticApiVersion )
-            .applicationId( applicationId )
+            .applicationId( applicationId == null ? null : applicationId.toString() )
             .channelAddress( channelAddress )
             .locale( locale )
             .build();

--- a/tools/samm-cli/src/test/java/org/eclipse/esmf/SammCliTest.java
+++ b/tools/samm-cli/src/test/java/org/eclipse/esmf/SammCliTest.java
@@ -1022,6 +1022,14 @@ class SammCliTest extends SammCliAbstractTest {
    }
 
    @Test
+   void testAspectToAsyncapiWithInvalidApplicationId() {
+      final ExecutionResult result = sammCli.apply( "--disable-color", "aspect", defaultInputFile, "to", "asyncapi", "-ai", "test" );
+      assertThat( result.exitStatus() ).isOne();
+      assertThat( result.stdout() ).isEmpty();
+      assertThat( result.stderr() ).contains( "Application id must be an absolute URI." );
+   }
+
+   @Test
    void testAspectToAsyncapiWithoutChannelAddress() {
       final ExecutionResult result = sammCli.apply( "--disable-color", "aspect", defaultInputFile, "to", "asyncapi" );
       assertThat( result.exitStatus() ).isZero();

